### PR TITLE
Feat(edit-user-info): Added User Update Form and Functionality

### PR DESCRIPTION
**Pull request by Noam6421**
### About:
This PR implements feature: [1235](https://dev.azure.com/spectrumFactory/CoronaI/_sprints/backlog/CoronaI%20Team/CoronaI/Sprint%2014?workitem=1235)

Major changes:
* Add Edit column to UserMangementTable.tsx that opens the EditUserInfoDialog.tsx.
* Added editUserInfo function to useUserManagment.tsx.
* Added EditUserInfoDialog.tsx component.
* Added new route put('/users) to useSignUpForm.tsx with new DB function that updates existing user by ID.
* Changed SignUpForm.tsx to take consider in EDIT mode and fixed auto-complete fields.
* Added validation to the form.

### CODE REVIEW

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx [hunk 1]
```diff
@@ -63,32 +62,45 @@ const GenericAlphabetTextField : React.FC<GenericAlphabetTextFieldProps> =
...
+        } else if (defaultValues?.investigationGroup?.id) {
+            fetchDesks(defaultValues.investigationGroup.id)
+        } else {
+            fetchDesks()
+        }
+    }, [watchCounty]);
+
+    const shouldDisableFields = mode === FormMode.READ ? true : false;
+
+    const shouldDisableEditFields = mode === FormMode.EDIT ? true : false;
```
Comment (y-golde) [inline client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx:94]: ternary ? is redundant

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx [hunk 2]
```diff
@@ -257,15 +270,18 @@ const SignUpForm: React.FC<Props> = ({ defaultValues, handleSaveUser, mode }: Pr
                         <FormInput xs={8} fieldName='שיוך ארגוני'>
                             <Controller
                                 name={SignUpFields.COUNTY}
-                                control={methods.control}
+                                control={methods.control}                                        
+                                defaultValue={defaultValues.investigationGroup ? defaultValues.investigationGroup : null}
```
Comment (y-golde) [inline client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx:274]: replace with : defaultValues.investigationGroup || null

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx [hunk 3]
```diff
@@ -257,15 +270,18 @@ const SignUpForm: React.FC<Props> = ({ defaultValues, handleSaveUser, mode }: Pr
...
                                 render={(props) => (
                                     <Autocomplete
+                                        {...props}
                                         disabled={shouldDisableFields}
                                         options={counties}
                                         getOptionLabel={(option) => option?.displayName}
                                         value={props.value}
                                         onChange={(event, selectedCounty) => {
-                                            props.onChange(selectedCounty ? selectedCounty.id : null)
+                                            props.onChange(selectedCounty ? selectedCounty : null)
```
Comment (y-golde) [inline client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx:283]: same here selectedCounty || null

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx [hunk 4]
```diff
@@ -285,15 +301,18 @@ const SignUpForm: React.FC<Props> = ({ defaultValues, handleSaveUser, mode }: Pr
                     <Grid item xs={4}>
                         <Controller
                             name={SignUpFields.DESK}
-                            control={methods.control}
+                            control={methods.control}                                    
+                            defaultValue={defaultValues.desk ? defaultValues.desk : null}
```
Comment (y-golde) [inline client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx:305]: same

File: client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts [hunk 5]
```diff
@@ -87,22 +119,52 @@ const useSignUp = ({ handleSaveUser }: useSignUpFormInCome) : useSignUpFormOutCo
...
+
+    const editUser = (updatedUser: SignUpUser) => {
+        const updateUserLogger = logger.setup('Update user');
+        updateUserLogger.info('launching updateUser request', Severity.LOW);
+        axios.put('/users', {
+            ...updatedUser, 
+            languages : updatedUser.languages || [], 
+            city : updatedUser.city?.id, 
+            investigationGroup : updatedUser.investigationGroup?.id,
+            desk : updatedUser.desk?.id ? updatedUser.desk?.id : null,
```
Comment (y-golde) [inline client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts:132]: export to function along with lines 107-112

**Review by y-golde — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx (hunk 1 above)
Comment (Noam6421) [reply client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx]: 👍🏼

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx (hunk 2 above)
Comment (Noam6421) [reply client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx]: 👍🏼

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx (hunk 3 above)
Comment (Noam6421) [reply client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx]: 👍🏼

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx (hunk 4 above)
Comment (Noam6421) [reply client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx:305]: 👍🏼

File: client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts (hunk 5 above)
Comment (Noam6421) [reply client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts]: 👍🏼

**Review by y-golde — APPROVED** [review]

File: client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx [hunk 6]
```diff
@@ -320,7 +339,8 @@ const SignUpForm: React.FC<Props> = ({ defaultValues, handleSaveUser, mode }: Pr
                                     <Autocomplete
                                         disabled={shouldDisableFields}
                                         options={sourcesOrganization}
-                                        getOptionLabel={(option) => option ? option.displayName : option}
+                                        getOptionLabel={(option) => option.displayName ? option.displayName : option}
```
Comment (chenFriedman) [inline client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx:342]: export (option) => option.displayName ? option.displayName : option} to const

File: client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts [hunk 7]
```diff
@@ -62,22 +63,58 @@ const useSignUp = ({ handleSaveUser }: useSignUpFormInCome) : useSignUpFormOutCo
                 alertError('לא ניתן היה לקבל שפות');
                 fetchLanguagesLogger.error('didnt get results back from the server', Severity.HIGH);    
             });
-    }
+    };
+    
+    const fetchDesks = (countyId?: number) => {            
+        const fetchDesksLogger = logger.setup('Getting desks');
+        if (!countyId) {
```
Comment (chenFriedman) [inline client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts:70]: maybe you can change the order of the if/else or export to const for avoiding the negative condition

**Review by chenFriedman — APPROVED** [review]

**Pull request merged**